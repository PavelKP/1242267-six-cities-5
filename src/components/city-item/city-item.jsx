import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../store/action';
import {cityPropTypes} from '../../prop-types/prop-types';

const CityItem = ({city, setCity, activeClass}) => {
  const handleCityClick = (evt) => {
    evt.preventDefault();
    setCity(city);
  };

  return (
    <li className="locations__item">
      <a className={`locations__item-link tabs__item ${activeClass}`}
        href="#"
        onClick={handleCityClick}>
        <span>{city.name}</span>
      </a>
    </li>
  );
};

CityItem.propTypes = {
  city: cityPropTypes.isRequired,
  setCity: PropTypes.func.isRequired,
  activeClass: PropTypes.string
};

const mapDispatchToProps = (dispatch) => ({
  setCity(city) {
    dispatch(ActionCreator.setCity(city));
  }
});

export {CityItem};
export default connect(null, mapDispatchToProps)(CityItem);
