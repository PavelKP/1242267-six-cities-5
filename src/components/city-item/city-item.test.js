import React from 'react';
import renderer from 'react-test-renderer';
import {CityItem} from './city-item';

const city = {
  name: `Moscow`,
  coordinates: [111, 222],
  zoom: 12,
};
const activeClass = `tabs__item--active`;


describe(`<CityItem /> render`, () => {

  it(`Should active CityItem render correctly`, ()=> {
    const tree = renderer.create(
        <CityItem
          city={city}
          activeClass={activeClass}
          setCity={()=>{}}
          setActiveSorting={()=>{}}
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should no-active CityItem render correctly`, ()=> {
    const tree = renderer.create(
        <CityItem
          city={city}
          setCity={()=>{}}
          setActiveSorting={()=>{}}
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

CityItem.defaultProps = {
  activeClass: ``,
};
