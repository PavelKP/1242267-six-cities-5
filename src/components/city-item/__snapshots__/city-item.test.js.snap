// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityItem /> render Should active CityItem render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      Moscow
    </span>
  </a>
</li>
`;

exports[`<CityItem /> render Should no-active CityItem render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    href="#"
    onClick={[Function]}
  >
    <span>
      Moscow
    </span>
  </a>
</li>
`;
